core/assertion.ts

```diff
@@ -78,13 +78,17 @@ export class AssertionContext {
     this.assertion = assertion;
   }
 
-  public ref(name: string) {
+  public ref(reference: string | IResolvable) {
+    const name =
+      typeof reference === "string" || typeof reference === "undefined"
```

Comment:
i don't think we should add explicit handling of `undefined` here - if anything such a value should throw (but I just wouldn't bother adding code explicitly for that case)

Reply:
Ref'ing `undefined` throws an error but it still adds the dependency to the list (see https://github.com/dataform-co/dataform/blob/58a5226eb5a8b57ae0e16bdd1852fb41a2ae6736/tests/core/core.spec.ts#L370).
Anyway, I've changed the code to make it look more elegant. Thanks.